src/metpy/plots/station_plot.py

```diff
@@ -336,6 +341,20 @@ def _plotting_units(u, v, plotting_units):
         v = np.array(v)
         return u, v
 
+    @staticmethod
+    def _scalar_plotting_units(scalar_value, plotting_units):
+        """Handle conversion to plotting units for barbs and arrows."""
+        if plotting_units:
+            if hasattr(scalar_value, 'units'):
+                scalar_value = scalar_value.to(plotting_units)
+            else:
+                raise ValueError('To convert to plotting units, units must be attached to '
+                                 'scalar value being converted.')
+
+        # Strip units, CartoPy transform doesn't like
```

Comment:
Stripping units shouldn't be necessary, CartoPy only hates them for barbs because it does the extra work of re-projecting the vectors to the transformed coordinate system.